orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -433,6 +432,34 @@ def _create_execution_agent(self):
 
         local_device_rank = self._device.index if device_type == "ort" else _utils.get_device_index(self._device)
 
+        if self._runtime_inspector.memory_ob.is_enabled() and self._debug_options.log_level <= LogLevel.INFO:
+            # Create a training agent without enabling memory optimization.
+            execution_agent = TrainingAgent(
```

Comment:
Can you elaborate why do we create this interim training agent?

Reply:
Done.